kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/LogCompactionWriterSuite.scala

```diff
@@ -0,0 +1,291 @@
+      s"""
+         |ALTER TABLE delta.`$path`
+         |SET TBLPROPERTIES
+         |('${TableFeatureProtocolUtils.propertyKey(DomainMetadataTableFeature)}' = 'enabled')
+         |""".stripMargin)
+
+    val deltaLog = DeltaLog.forTable(spark, path)
+    val domainMetadata = DeltaSparkDomainMetadata("testDomain1", d1Val, false) ::
+      DeltaSparkDomainMetadata("testDomain2", d2Val, false) :: Nil
+    deltaLog.startTransaction().commit(domainMetadata, Truncate())
+  }
+
+  Seq(false, true).foreach { includeRemoves =>
+    Seq(false, true).foreach { includeDM =>
+      val removesMsg = if (includeRemoves) " and removes" else ""
+      val dmMsg = if (includeDM) ", include multiple PandM and DomainMetadata" else ""
+      test(s"commits containing adds${removesMsg}${dmMsg}") {
+        withTempDirAndEngine { (tablePath, engine) =>
+          addData(tablePath, alternateBetweenAddsAndRemoves = includeRemoves, numberIter = 8)
+          if (includeDM) {
+            addDomainMetadata(tablePath, "", "{\"key1\":\"value1\"}")
+            addDomainMetadata(tablePath, "here", "{\"key2\":\"value2\"}")
+          }
+
+          val expectedLastCommit = if (includeDM) {
+            11 // 0-7 for add/removes + 2 for enable+set DomainMetatdata
+          } else {
+            7 // 0-7 for add/removes
+          }
+
+          val actionsFromCommits =
+            getActionsFromLog(new Path(tablePath), engine, 0, expectedLastCommit)
+
+          val dataPath = new Path(s"file:${tablePath}")
+          val logPath = new Path(s"file:${tablePath}", "_delta_log")
+
+          val hook = new LogCompactionHook(
+            dataPath,
+            logPath,
+            0,
+            expectedLastCommit,
+            0)
+          hook.threadSafeInvoke(engine)
+          val endCommitStr = f"$expectedLastCommit%020d"
+          val compactedPath =
+            tablePath + s"/_delta_log/00000000000000000000.${endCommitStr}.compacted.json"
+          val actionsFromCompacted = getActionsFromCompacted(compactedPath, engine)
+
+          checkAnswer(actionsFromCompacted, actionsFromCommits)
+        }
+      }
+    }
+  }
+
+  Seq(false, true).foreach { includeRemoves =>
+    val testMsgUpdate = if (includeRemoves) " and removes" else ""
+    test(s"Read table with adds$testMsgUpdate") {
+      withTempDirAndEngine { (tablePath, engine) =>
+        addData(tablePath, alternateBetweenAddsAndRemoves = includeRemoves, numberIter = 10)
+
+        spark.conf.set(DeltaSQLConf.DELTALOG_MINOR_COMPACTION_USE_FOR_READS.key, "false")
+        val withoutCompactionData = readUsingSpark(tablePath)
+
+        val dataPath = new Path(s"file:${tablePath}")
+        val logPath = new Path(s"file:${tablePath}", "_delta_log")
+        val hook = new LogCompactionHook(dataPath, logPath, 0, 9, 0)
+        hook.threadSafeInvoke(engine)
+
+        spark.conf.set(DeltaSQLConf.DELTALOG_MINOR_COMPACTION_USE_FOR_READS.key, "true")
+        val withCompactionData = readUsingSpark(tablePath)
```

Comment:
Only an idea but if you include domain metadata you can check those with spark as well by loading the snapshot and accessing `snapshot.domainMetadata`

Reply:
`DeltaLog.forTable(spark, tablePath).getSnapshotAt(version)` to get the Spark snapshot in Kernel btw